hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/replication/TestContainerImporter.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos;
+import org.apache.hadoop.hdds.scm.container.common.helpers.StorageContainerException;
+import org.apache.hadoop.ozone.container.common.helpers.ContainerUtils;
+import org.apache.hadoop.ozone.container.common.impl.ContainerData;
+import org.apache.hadoop.ozone.container.common.impl.ContainerDataYaml;
+import org.apache.hadoop.ozone.container.common.impl.ContainerLayoutVersion;
+import org.apache.hadoop.ozone.container.common.impl.ContainerSet;
+import org.apache.hadoop.ozone.container.common.volume.MutableVolumeSet;
+import org.apache.hadoop.ozone.container.common.volume.StorageVolume;
+import org.apache.hadoop.ozone.container.keyvalue.KeyValueContainer;
+import org.apache.hadoop.ozone.container.keyvalue.KeyValueContainerData;
+import org.apache.hadoop.ozone.container.ozoneimpl.ContainerController;
+import org.apache.ozone.test.GenericTestUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.apache.hadoop.ozone.container.replication.CopyContainerCompression.NO_COMPRESSION;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for {@link ContainerImporter}.
+ */
+class TestContainerImporter {
+
+  private OzoneConfiguration conf;
+
+  @BeforeEach
+  void setup() {
+    conf = new OzoneConfiguration();
+  }
+
+  @Test
+  void importSameContainerWhenAlreadyImport() throws Exception {
+    long containerId = 1;
+    // create container
+    KeyValueContainerData containerData = new KeyValueContainerData(containerId,
+        ContainerLayoutVersion.FILE_PER_BLOCK, 100, "test", "test");
+    KeyValueContainer container = new KeyValueContainer(containerData, conf);
+    ContainerController controllerMock = mock(ContainerController.class);
+    // create containerImporter object
+    ContainerSet containerSet = new ContainerSet(0);
+    containerSet.addContainer(container);
+    MutableVolumeSet volumeSet = new MutableVolumeSet("test", conf, null,
+        StorageVolume.VolumeType.DATA_VOLUME, null);
+    ContainerImporter containerImporter = new ContainerImporter(conf,
+        containerSet, controllerMock, volumeSet);
+    File tarFile = new File("dummy.tar");
+    // second import should fail immediately
+    try {
+      containerImporter.importContainer(containerId, tarFile.toPath(),
+          null, NO_COMPRESSION);
+      assertFalse(true, "exception should occur");
+    } catch (StorageContainerException ex) {
+      assertTrue(ex.getResult().equals(
+          ContainerProtos.Result.CONTAINER_EXISTS));
+    }
```

Comment:
Nit: can be simplified.

```suggestion
    StorageContainerException ex = assertThrows(StorageContainerException.class,
        () -> containerImporter.importContainer(containerId, tarFile.toPath(),
            null, NO_COMPRESSION));
    assertEquals(ContainerProtos.Result.CONTAINER_EXISTS, ex.getResult());
```

Applies to the other test method, too.

Reply:
@adoroszlai Thanks, all changes are updated

Reply:
@sumitagrawl did you forget to push the update to your fork?

Reply:
@adoroszlai sorry, forget to push, done now